source/web-service/flaskapp/routes/records.py

```diff
@@ -540,6 +550,69 @@ def entity_record(entity_id):
             return abort(response)
 
 
+# 'DELETE' method.
+@records.route("/<path:id>", methods=["DELETE"])
+def delete(id):
+
+    # Authentication
+    status = authenticate_bearer(request)
+    if status != status_ok:
+        response = construct_error_response(status)
+        return abort(response)
+
+    current_app.logger.debug("Authentication checked - DELETE request allowed.")
+
+    # Get record from DB
+    db_rec = get_record(id)
+
+    # No such record or a stub record
+    if db_rec is None or (db_rec.data is None and db_rec.checksum is None):
+        response = construct_error_response(status_record_not_found)
+        return abort(response)
+
+    # Process DELETE
+    with db.session.no_autoflush:
+        try:
+            current_app.logger.debug(f"Starting delete process on {id}")
+
+            process_activity(db_rec.id, Event.Delete)
+            record_delete(db_rec, None, True)
```

Comment:
Should be record_delete(db_rec, None) - don't commit at this stage as it will commit this and the activity to the DB before this has checked that it can be removed from the graphstore.

Reply:
(Had to comment this again as github decided to hide it for some reason? The original is in that hidden comment section)

Reply:
Fixed